src/Mono.Android/Test/Xamarin.Android.Net/AndroidClientHandlerTests.cs

```diff
@@ -151,15 +151,37 @@ protected Task<HttpResponseMessage> ConnectIgnoreFailure (Func<Task<HttpResponse
 			try {
 				return connector ();
 			} catch (AggregateException ex) {
-				if (ex.InnerException is WebException wex && wex.Status == WebExceptionStatus.ConnectFailure) {
-					connectionFailed = true;
-					Assert.Ignore ($"Failed to connect to server. {wex}");
+				if (IgnoreIfConnectionFailed (ex.InnerException as WebException, out connectionFailed))
 					return null;
-				}
+				throw;
+			}
+		}
 
+		protected void RunIgnoringNetworkIssues (Action runner, out bool connectionFailed)
+		{
+			connectionFailed = false;
+			try {
+				runner ();
+			} catch (AggregateException ex) {
+				if (IgnoreIfConnectionFailed (ex.InnerException as WebException, out connectionFailed))
+					return;
 				throw;
 			}
 		}
+
+		bool IgnoreIfConnectionFailed (WebException wex, out bool connectionFailed)
```

Comment:
Why does this need an `out bool` parameter, and/or why does this need a `bool` return value?  As far as I can tell, every codepath has `connectionFailed==return-value`.

Reply:
The `out bool` is to stay DRY - this function is called from two different methods which have `out bool conectionFailed` used by **their** caller. And the return `bool` is to indicate whether to ignore the exception thrown or rethrow it.